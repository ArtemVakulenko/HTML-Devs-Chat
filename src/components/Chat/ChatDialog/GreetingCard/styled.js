import styled from 'styled-components';

export const StGreetingCardContainer = styled.div`
  height: 100%;
  width: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
`;
export const StGreetingCardMessage = styled.div`
  width: 70%;
  font-size: 3rem;
  text-align: center;
`;
