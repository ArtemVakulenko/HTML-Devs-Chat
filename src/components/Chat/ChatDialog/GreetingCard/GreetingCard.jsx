import React from 'react';
import PropTypes from 'prop-types';
import { useTranslation } from 'react-i18next';
import { StGreetingCardContainer, StGreetingCardMessage } from './styled';

const GreetingCard = ({ userName }) => {
  const { t } = useTranslation();
  const hello = t('Hello');
   const greeting = t('Greeting');
  return (
    <StGreetingCardContainer>
      <StGreetingCardMessage>
        {hello}
        {userName}
        {greeting}
      </StGreetingCardMessage>
    </StGreetingCardContainer>
  );
};

GreetingCard.propTypes = {
  userName: PropTypes.string,
};

export default GreetingCard;
