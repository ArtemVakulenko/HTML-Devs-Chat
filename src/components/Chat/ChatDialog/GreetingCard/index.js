import { connect } from 'react-redux';
import { userName } from 'store/mainPage/selectors';
import GreetingCard from './GreetingCard';

const mapStateToProps = (state) => ({
  userName: userName(state),
});

export default connect(mapStateToProps, null)(GreetingCard);
