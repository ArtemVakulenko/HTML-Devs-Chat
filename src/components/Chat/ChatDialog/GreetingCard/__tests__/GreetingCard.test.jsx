import React from 'react';
import GreetingCard from '../GreetingCard';

jest.mock('react-i18next', () => ({
    useTranslation: jest.fn().mockImplementation(() => ({
      t: jest.fn(),
    })),
  }));
  jest.mock('../../../../hooks', () => ({
    useTheme: jest.fn().mockImplementation(() => ({
      colors: {
        dark: {
          color: '',
          input: '',
          placeholder: '',
        },
      },
      theme: 'dark',
    })),
  }));

  describe('ChatHeader', () => {
    it('Should match snapshot', () => {
      const component = global.shallow(<GreetingCard/>);
      expect(component.html()).toMatchSnapshot();
    });
  });
