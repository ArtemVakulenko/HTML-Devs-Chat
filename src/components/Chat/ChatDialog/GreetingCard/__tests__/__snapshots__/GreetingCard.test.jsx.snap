// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatHeader Should match snapshot 1`] = `"<div class=\\"styled__StGreetingCardContainer-GpdgC dvsMpA\\"><div class=\\"styled__StGreetingCardMessage-hzfsPu iaAThl\\"></div></div>"`;
